homeassistant/components/workday/__init__.py

```diff
@@ -1 +1,128 @@
+import logging
+from typing import Any
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_COUNTRY, CONF_NAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.helpers.typing import ConfigType
+
+from . import util
+from .binary_sensor import WorkdayBinarySensor
+from .const import (
+    ALLOWED_DAYS,
+    CONF_ADD_HOLIDAYS,
+    CONF_EXCLUDES,
+    CONF_OFFSET,
+    CONF_PROVINCE,
+    CONF_REMOVE_HOLIDAYS,
+    CONF_WORKDAYS,
+    DEFAULT_EXCLUDES,
+    DEFAULT_OFFSET,
+    DEFAULT_WORKDAYS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Handle Workday setup from YAML."""
+    # This used to be under the binary_sensor platform, now it's its own platform
+    if "binary_sensor" not in config:
+        return True
+
+    for sensor_config in config["binary_sensor"]:
+        if sensor_config["platform"] != "workday":
+            continue
+
+        workday_data: dict[str, Any] = {
+            CONF_NAME: sensor_config["name"],
+            CONF_COUNTRY: sensor_config["country"],
+            CONF_PROVINCE: sensor_config.get("province", ""),
+            CONF_WORKDAYS: sensor_config.get("workdays", DEFAULT_WORKDAYS),
+            CONF_EXCLUDES: sensor_config.get("excludes", DEFAULT_EXCLUDES),
+            CONF_OFFSET: int(sensor_config.get("days_offset", DEFAULT_OFFSET)),
+        }
+
+        for day in workday_data[CONF_WORKDAYS] + workday_data[CONF_EXCLUDES]:
+            assert day in ALLOWED_DAYS
+
+        add_holidays = sensor_config.get("add_holidays", [])
+        remove_holidays = sensor_config.get("remove_holidays", [])
+
+        workday_data[CONF_ADD_HOLIDAYS] = ",".join(
+            sorted([d.strftime("%Y-%m-%d") for d in add_holidays])
+        )
+        workday_data[CONF_REMOVE_HOLIDAYS] = ",".join(
+            sorted([d.strftime("%Y-%m-%d") for d in remove_holidays])
+        )
+
+        _LOGGER.debug("Importing Workday config from YAML: %s", workday_data)
+
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data=workday_data,
+            )
+        )
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml",
+        breaks_in_ha_version="2023.5.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Workday from a config entry."""
+
+    _LOGGER.debug(
+        "Setting up a new Workday entry: %s", util.config_entry_to_string(entry)
+    )
```

Comment:
I think there is way too many debug statements around. I think they might be useful internally to debug the code, but I don't think they should be in the PR.